cli/lsp/tsc.rs

```diff
@@ -2124,4 +2143,88 @@ mod tests {
       serde_json::from_value(result.unwrap()).unwrap();
     assert!(response.is_some());
   }
+
+  #[test]
+  fn test_modify_sources() {
+    let (mut runtime, state_snapshot, location) = setup(
+      true,
+      json!({
+        "target": "esnext",
+        "module": "esnext",
+        "lib": ["deno.ns", "deno.window"],
+        "noEmit": true,
+      }),
+      vec![(
+        "file:///a.ts",
+        r#"
+          import * as a from "https://deno.land/x/example/a.ts";
+          if (a.a === "b") {
+            console.log("fail");
+          }
+        "#,
+        1,
+      )],
+    );
+    let cache = HttpCache::new(&location);
+    let specifier_dep =
+      resolve_url("https://deno.land/x/example/a.ts").unwrap();
+    cache
+      .set(
+        &specifier_dep,
+        Default::default(),
```

Comment:
```suggestion
        Url::default(),
```
Makes it immediately obvious what's being default-constructed. Ditto on line 2211.